Adafruit_AM2315.cpp

```diff
@@ -75,8 +75,10 @@ boolean Adafruit_AM2315::readData(void) {
   //
 
   // Step one: Wake-up sensor
-  // 10 * 8 clock cycles @ 100kHz ~= 800us
-  i2c_dev->write(reply, 10);
+  // Datasheet is confusing about "wake up". This is adapted from:
+  // https://www.switchdoc.com/2016/04/am2315-highly-reliable-esp8266-arduino-driver/
+  i2c_dev->write(reply, 1);
+  delay(50);
```

Comment:
The datasheet says to wait 800us to 3ms. 50 seems a bit much. I tested with 1ms and it worked

Reply:
Thanks. That's in reference to this sensor wake up from the [datasheet](https://cdn-shop.adafruit.com/datasheets/AM2315.pdf), section 7.2.4:
![image](https://user-images.githubusercontent.com/8755041/153017753-faacc788-31f1-4172-bb02-b055adeb9e00.png)
which is what the original code was attempting to do. The diagram is more useful than the confusing text. It shows the delay of 800us-3ms being before the I2C stop occurs. And with **both** SCL and SDA being held low for that period. The SDA part is easy, just clock out `0`s. The SCL part is not, at least with most I2C interfaces.

So what we're doing with this PR is something not even documented in the datasheet. Putting a lot of faith in that linked webpage. It appears they did a lot of talking directly with the manufacturer. So maybe that `delay(50)` came out of that? Or they just made a guess?

> if the host is hardware I2C, you do not need to wait

So maybe we don't even need the delay at all? Just a very confusing description.

I think let's stick with what they used for now just the sake of consistency. Can look at things further if this update can be demonstrated to fail.

Reply:
Yeah, you’re right. Let’s go with this for now. I’ll try running the sensor over some time to test.